proposals/XXXX-efficient-token-program.md

```diff
@@ -0,0 +1,78 @@
+---
+simd: 'XXXX'
+title: Efficient Token program
```

Comment:
nit: May as well include "P-Token" in the title if we're confident with the name :smile: